bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java

```diff
@@ -89,20 +102,65 @@ public static Optional<AutoScaleMethod> forString(String s) {
 
 	static {
 		autoScaleValue = System.getProperty (SWT_AUTOSCALE);
-
+		setUseSmoothScalingByDefaultProvider(() -> isMonitorSpecificScalingActive());
```

Comment:
This is not correct, as it will erroneously set that behavior for MacOS and Linux as well.